pkg/server/endpoints/middleware.go

```diff
@@ -104,6 +105,9 @@ func AgentAuthorizer(log logrus.FieldLogger, ds datastore.DataStore, clk clock.C
 	return middleware.AgentAuthorizerFunc(func(ctx context.Context, agentID spiffeid.ID, agentSVID *x509.Certificate) error {
 		id := agentID.String()
 		log := log.WithField(telemetry.AgentID, id)
+		if !rpccontext.CallerIsLocal(ctx) {
```

Comment:
I was about to make a comment that the caller address is already added to the logger attached to the request context by the authorization middleware before the agent authorizer is called, but then noticed that the `AgentAuthorizer` is using the `log` that is passed in. I think ideally what we should do here is remove the `log` parameter from `AgentAuthorizer` and instead use `rpccontext.Logger(ctx)` to get at the logger to use. Both the caller ID and address will already be added to that logger so we should need to those fields inside this function.

Reply:
Fixed https://github.com/spiffe/spire/pull/2281/commits/80941355b565c51a0f68b4fa73077c215a482466
I'm afraid changing field name from `agent_id` to `caller_id`. Do I worried too much?

Reply:
Typically we do exercise some caution when adjusting field names, so you are right to be worried :)

However, the maintainers talked this over and feel like the change is justified in the name of increased consistency. The `agent_id` field is only used in one spot outside of this, and that is for the field containing the agent ID produced during attestation. Everywhere else related to incoming caller SPIFFE ID uses `caller_id`.